cmd/state-installer/cmd.go

```diff
@@ -482,3 +476,40 @@ func determineLegacyUpdate(stateToolInstalled bool, packagedStateExe, payloadPat
 func noArgs() bool {
 	return len(os.Args[1:]) == 0
 }
+
+func shouldUpdateInstalledStateTool() bool {
+	logging.Debug("Checking if installed state tool is an older version.")
+
+	stateExe := constants.StateCmd + exeutils.Extension
+	cmd := exec.Command(stateExe, "--version")
+	cmd.Env = os.Environ()
+	output, err := cmd.Output()
+	if err != nil {
+		logging.Debug("Could not determine state tool version.")
+		return true // probably corrupted install
+	}
+
+	branchRegex := regexp.MustCompile("Branch (\\S+)")
```

Comment:
Please move this to the global space so that it trips at startup rather than at this point in the control flow. Generally, "must" functions are best defined at a point where they will trip as soon as possible.

Reply:
Good to know, thanks.